pkg/ssl.go

```diff
@@ -0,0 +1,204 @@
+package pkg
+
+import (
+	"encoding/pem"
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"github.com/wttech/aemc/pkg/common/cryptox"
+	"github.com/wttech/aemc/pkg/common/filex"
+	"github.com/wttech/aemc/pkg/common/osx"
+	"github.com/wttech/aemc/pkg/common/pathx"
+	"golang.org/x/net/html"
+	"io"
+	"os"
+	"strings"
+)
+
+const (
+	SSLSetupPath = "/libs/granite/security/post/sslSetup.html"
+)
+
+type SSL struct {
+	instance *Instance
+}
+
+type sslLock struct {
+	KeystorePassword   string `yaml:"keystore_password"`
+	TrustStorePassword string `yaml:"trust_store_password"`
+	Certificate        string `yaml:"certificate"`
+	PrivateKey         string `yaml:"private_key"`
+	HttpsHostname      string `yaml:"https_hostname"`
+	HttpsPort          string `yaml:"https_port"`
+}
+
+func NewSSL(instance *Instance) *SSL {
+	return &SSL{
+		instance: instance,
+	}
+}
+
+func (s SSL) Setup(keyStorePassword, trustStorePassword, certificateFile, privateKeyFile, httpsHostname, httpsPort string) (bool, error) {
+	if !pathx.Exists(certificateFile) {
+		return false, fmt.Errorf("%s > certificate file does not exist: %s", s.instance.ID(), certificateFile)
+	}
+	if !pathx.Exists(privateKeyFile) {
+		return false, fmt.Errorf("%s > private key file does not exist: %s", s.instance.ID(), privateKeyFile)
+	}
+
+	privateKeyData, err := os.ReadFile(privateKeyFile)
+	if err != nil {
+		return false, fmt.Errorf("%s > failed to read private key file: %w", s.instance.ID(), err)
+	}
+	pemBlock, _ := pem.Decode(privateKeyData)
+	if pemBlock != nil {
+		tempDerFile, err := os.CreateTemp("", "aemc-private-key-*.der")
+		if err != nil {
+			return false, fmt.Errorf("%s > failed to create temp file for storing DER certificate: %w", s.instance.ID(), err)
+		}
+		defer os.Remove(tempDerFile.Name())
+		err = writeDER(tempDerFile, pemBlock)
+		if err != nil {
+			return false, fmt.Errorf("%s > failed to write DER certificate: %w", s.instance.ID(), err)
+		}
+		privateKeyFile = tempDerFile.Name()
+	}
+
+	lock := s.lock(keyStorePassword, trustStorePassword, certificateFile, privateKeyFile, httpsHostname, httpsPort)
+	check, err := lock.State()
+	if err != nil {
+		return false, fmt.Errorf("%s > failed to check SSL setup: %w", s.instance.ID(), err)
```

Comment:
this messages hides what's going on exactly.

you are not always obligated to add message to error 
you could return it as is ...
```
err := ...
if err != nil {
  return err
}
```